src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -125,17 +127,27 @@ public void prepareRun(BatchSinkContext context) throws Exception {
       null : GCPUtils.loadServiceAccountCredentials(config.connection.getServiceAccount(), isServiceAccountFilePath);
     Storage storage = GCPUtils.getStorage(config.connection.getProject(), credentials);
     Bucket bucket;
+    String location;
     try {
       bucket = storage.get(config.getBucket());
+      location = bucket.getLocation();
     } catch (StorageException e) {
       throw new RuntimeException(
         String.format("Unable to access or create bucket %s. ", config.getBucket())
           + "Ensure you entered the correct bucket path and have permissions for it.", e);
     }
     if (bucket == null) {
       GCPUtils.createBucket(storage, config.getBucket(), config.getLocation(), cmekKeyName);
+      location = config.getLocation();
     }
     this.outputPath = getOutputDir(context);
+    // create asset for lineage
+    String referenceName = Strings.isNullOrEmpty(config.getReferenceName())
+      ? ReferenceNames.normalizeFqn(config.getPath())
+      : config.getReferenceName();
+    asset = Asset.builder(referenceName)
+      .setFqn(config.getPath()).setLocation(location).build();
+    super.prepareRun(context);
```

Comment:
Add a comment here that states why prepare run is called at the end (to ensure this doesn't get moved to the top by accident).